### PR TITLE
more automation for grasping

### PR DESCRIPTION
**Pull request by manuelli**
- adds more functionality to grasp_supervisor
- some bug fixes
- reverts unintentional breaking changes to `pub_joint_state.py` that snuck in during #212.

This contains some important bug fixes so I would merge as soon as the build passes @peteflorence. This has been tested on the robot.

---
This change is [Reviewable](https://reviewable.io/reviews/robotlocomotion/spartan/221)

### CODE REVIEW

**Comment by gizatt** [comment]
Review status: 0 of 14 files reviewed at latest revision, 1 unresolved discussion.

---

*[setup/docker/spartan.dockerfile, line 46 at r1](https://reviewable.io:443/reviews/robotlocomotion/spartan/221#-L0pkf-ckga3WHIjljsk:-L0pkf-ckga3WHIjljsl:blhypao) ([raw file](https://github.com/robotlocomotion/spartan/blob/fc82a3ce41629fa71613e871e4e8ad9d1df524c4/setup/docker/spartan.dockerfile#L46)):*
> ```Dockerfile
> 
> # change ownership of everything to our user
> cd $WORKDIR
> ```

This line is giving me trouble:

```Error response from daemon: Dockerfile parse error line 46: unknown instruction: CD```

Switching to

```
RUN cd $WORKDIR && chown $USER_NAME:$USER_NAME -R .
```

appears to help
```


---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/spartan/221)*
<!-- Sent from Reviewable.io -->

**Comment by gizatt** [comment]
I'm OK with the state of this (and am fine with having it merged), but how do I test it? Are there any self-contained scripts or entrypoints I should look at to try it on the robot?

**Comment by manuelli** [comment]
Review status: 0 of 14 files reviewed at latest revision, 1 unresolved discussion.

---

*[setup/docker/spartan.dockerfile, line 46 at r1](https://reviewable.io:443/reviews/robotlocomotion/spartan/221#-L0pkf-ckga3WHIjljsk:-L0qLrA3upHoI4kPtgjU:b-mq96y1) ([raw file](https://github.com/robotlocomotion/spartan/blob/fc82a3ce41629fa71613e871e4e8ad9d1df524c4/setup/docker/spartan.dockerfile#L46)):*
<details><summary><i>Previously, gizatt (Gregory Izatt) wrote…</i></summary><blockquote>

This line is giving me trouble:

```Error response from daemon: Dockerfile parse error line 46: unknown instruction: CD```

Switching to

```
RUN cd $WORKDIR && chown $USER_NAME:$USER_NAME -R .
```

appears to help
</blockquote></details>

Ok that sounds like the correct thing to do. Sorry for the bug on my part. If you could go ahead and commit that it would be great. I am currently out of the country on my personal laptop that can't run nvidia-docker.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/spartan/221)*

**Comment by manuelli** [comment]
Hey Greg, thanks for catching that dockerfile bug. As for on robot testing I have tested the grasping stuff pretty well and I think it works. Unfortunately it's a bit tied up with my `spartan_grasp` repo which has to run in a separate docker etc. But you could try launching director and then in the terminal

```
graspSupervisor.testCollectSensorData()
```

which should move the arm to two poses and capture the pointclouds.

**Comment by gizatt** [comment]
Understood -- I'll give it a whirl tomorrow (more motivated by maintaining some personal understanding rather than anything else) and work with Pete to get this (and the other PRs) merged before the holiday.

---

Review status: 0 of 14 files reviewed at latest revision, 1 unresolved discussion.

---

*[setup/docker/spartan.dockerfile, line 46 at r1](https://reviewable.io:443/reviews/robotlocomotion/spartan/221#-L0pkf-ckga3WHIjljsk:-L0qP81V0C98-wNf2MDm:b-4hvi0z) ([raw file](https://github.com/robotlocomotion/spartan/blob/fc82a3ce41629fa71613e871e4e8ad9d1df524c4/setup/docker/spartan.dockerfile#L46)):*
<details><summary><i>Previously, manuelli (Lucas Manuelli) wrote…</i></summary><blockquote>

Ok that sounds like the correct thing to do. Sorry for the bug on my part. If you could go ahead and commit that it would be great. I am currently out of the country on my personal laptop that can't run nvidia-docker.
</blockquote></details>

Got it!

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/spartan/221#-:-L0qPD9Mj-s7nVKLnwEK:b2wi88u)*

File: setup/docker/spartan.dockerfile [hunk 1]
```diff
@@ -43,7 +43,7 @@ COPY ./setup/docker/terminator_config .config/terminator/config
 RUN chown $USER_NAME:$USER_NAME -R .config
 
 # change ownership of everything to our user
-RUN chown $USER_NAME:$USER_NAME -R .
+RUN cd $WORKDIR && chown $USER_NAME:$USER_NAME -R .
```
Comment (peteflorence) [inline setup/docker/spartan.dockerfile]: Why was this needed?

File: src/catkin_projects/robot_msgs/srv/RunIK.srv [hunk 2]
```diff
@@ -1,4 +1,6 @@
 geometry_msgs/PoseStamped pose_stamped
+sensor_msgs/JointState[] seed_pose # optional seed pose
+sensor_msgs/JointState[] nominal_pose # optional nominal pose
```
Comment (peteflorence) [inline src/catkin_projects/robot_msgs/srv/RunIK.srv]: These are great

File: src/catkin_projects/robot_server/src/robot_server/pub_joint_state.py [hunk 3]
```diff
@@ -82,17 +64,13 @@ def onIiwaStatus(self, channel, data):
...
         self.lc.handle_timeout(0.005)
-        #print "LOOPING INSIDE pub_joint_state"
-        #print ""
+        self.publishROSJointStateMessage()
 
-    def publishROSJointStateMessage(self, ros_time_now_previously_grabbed):
+    def publishROSJointStateMessage(self):
         self.robot_state.header = Header()
-        self.robot_state.header.stamp = ros_time_now_previously_grabbed
+        self.robot_state.header.stamp = rospy.Time.now()
```
Comment (peteflorence) [inline src/catkin_projects/robot_server/src/robot_server/pub_joint_state.py]: agreed this file should all be reverted, also nice that will not be needed for hardware due to https://github.com/RobotLocomotion/spartan/pull/215

**Comment by peteflorence** [comment]
LGTM

File: setup/docker/spartan.dockerfile (hunk 1 above)
Comment (peteflorence) [reply setup/docker/spartan.dockerfile]: ah, I see Greg's comment
